src/libopenrave/kinbody.cpp

```diff
@@ -6287,13 +6289,6 @@ UpdateFromInfoResult KinBody::UpdateFromKinBodyInfo(const KinBodyInfo& info)
         }
         updateFromInfoResult = UFIR_Success;
     }
-    if( info._vLinkInfos.size() > 0 ) {
```

Comment:
`_baseLinkInBodyTransform` and `_invBaseLinkInBodyTransform` are part of the input `KinBodyInfo` and it is incorrect to leave them not updated.

Reply:
This is where the bug come from. The `_baseLinkInBodyTransform` is the relative transformation of link to the body, and is simply not equal to the transformation of the first link.

Reply:
Yes, I understand that `_baseLinkInBodyTransform` should be the relative transform you mentioned.

When `KinBodyInfo` is generated (supposedly from `KinBody::ExtractInfo`), the kinbody transform is set to 
identity first before link infos are extracted. So `info._vLinkInfos[0]->GetTransform()` should, in fact, represent the `_baseLinkInBodyTransform`.